src/test/groovy/com/commercehub/gradle/plugin/avro/OptionsFunctionalSpec.groovy

```diff
@@ -147,6 +148,7 @@ class OptionsFunctionalSpec extends FunctionalSpec {
         "'false'"       | false
     }
 
+    @Ignore("Velocity error, for some reason \$velocityCount is unset in a foreach")
```

Comment:
I've added this ignore because this test is now failing. I don't know why.

Reply:
The error I get is "Variable $velocityCount has not been set at /alternateTemplates/record.vm[line 73, column 150]", which seems to indicate that there was a variable previously passed in to the templates (and used by my test template) that no longer is.

If you grab an updated copy of the template from https://github.com/apache/avro/blob/master/lang/java/compiler/src/main/velocity/org/apache/avro/compiler/specific/templates/java/classic/record.vm, put it in the project at /src/test/resources/com/commercehub/gradle/plugin/avro/record.vm, and retain the `// Custom template` comment after the class declaration, it should let the test pass in Avro 1.9.

It might introduce compatibility errors with other Avro versions, however.  You can check this using `./gradlew testVersionCompatibility`.  If so, making some minimal changes to the template to eliminate use of "$velocityCount" is probably the way to go.

Reply:
The updated `record.vm` does fix that test.

However, note that `testVersionCompatibility` has many failures on my machine, but none related to this test, and with or without the changes to `record.vm`. I'll update the `record.vm` and unignore the test, as it isn't making anything worse.

Reply:
Thanks for making that update.  If you have time, I'd be interested in hearing more about these failures in `testVersionCompatibility` on your machine, as they're not something that I'm currently able to reproduce, and I'm interested in trying to figure out the causes of the differences.

Reply:
Let me know if you want me to look at `testVersionCompatibility` but right now its not clear to me how to fix this, so I'll need some pointers.

Reply:
If you could send me the output of `./gradlew --version` and your test results HTML report (build/reports/tests/test/** attached as a ZIP file), that should be enough for me to know what (if any) questions to ask next.

Reply:
It wonder if it might have something to do with the mismatch between the Avro dependency in the buildscript block, and the compile dependency.

This is a build file generated by one of the failing tests:

```
            buildscript {
                dependencies {
                    classpath files('/home/raman/source_ext/gradle-avro-plugin/build/classes/java/main', '/home/raman/source_ext/gradle-avro-plugin/build/classes/groovy/main', '/home/raman/source_ext/gradle-avro-plugin/build/resources/main', '/home/raman/.gradle/caches/4.10.2/generated-gradle-jars/gradle-api-4.10.2.jar', '/home/raman/.gradle/wrapper/dists/gradle-4.10.2-bin/cghg6c4gf4vkiutgsab8yrnwv/gradle-4.10.2/lib/groovy-all-2.4.15.jar', '/home/raman/.gradle/wrapper/dists/gradle-4.10.2-bin/cghg6c4gf4vkiutgsab8yrnwv/gradle-4.10.2/lib/gradle-installation-beacon-4.10.2.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.apache.avro/avro-compiler/1.9.0-SNAPSHOT/e69ea37c403bbe05185ee9b31d5f6b60a0dc9e7a/avro-compiler-1.9.0-SNAPSHOT.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.apache.avro/avro/1.9.0-SNAPSHOT/17a984a8605655c1e5b74d6e6c919c3078f29ec6/avro-1.9.0-SNAPSHOT.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.apache.velocity/velocity-engine-core/2.0/6e5f29e1237b1764a4ce769feeffb85b0b19cfa7/velocity-engine-core-2.0.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-lang3/3.8.1/6505a72a097d9270f7a9e7bf42c4238283247755/commons-lang3-3.8.1.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-databind/2.9.8/11283f21cc480aa86c4df7a0a3243ec508372ed2/jackson-databind-2.9.8.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/joda-time/joda-time/2.10.1/9ac3dbf89dbf2ee385185dd0cd3064fe789efee0/joda-time-2.10.1.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.slf4j/slf4j-api/1.7.25/da76ca59f6a57ee3102f8f9bd9cee742973efa8a/slf4j-api-1.7.25.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-core/2.9.8/f5a654e4675769c716e5b387830d19b501ca191/jackson-core-2.9.8.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/org.apache.commons/commons-compress/1.18/1191f9f2bc0c47a8cce69193feb1ff0a8bcb37d5/commons-compress-1.18.jar', '/home/raman/.gradle/caches/modules-2/files-2.1/com.fasterxml.jackson.core/jackson-annotations/2.9.0/7c10d545325e3a6e72e06381afe469fd40eb701/jackson-annotations-2.9.0.jar')
                }
            }
            repositories {
                jcenter()
                maven {
                    url 'https://repository.apache.org/content/repositories/snapshots/'
                }
            }
        apply plugin: "com.commercehub.gradle.plugin.avro"
dependencies { compile "org.apache.avro:avro:1.8.2" }
```